config/v1/types_ingress.go

```diff
@@ -172,6 +180,98 @@ type ComponentRouteStatus struct {
 	RelatedObjects []ObjectReference `json:"relatedObjects"`
 }
 
+// RequiredRouteAnnotations specifies annotations that are required to be set on
+// newly created routes matching some criteria.
+type RequiredRouteAnnotations struct {
+	// domains is an optional list of domains for which these annotations
+	// are required.  If domains is specified and a route is created with a
+	// spec.host matching one of the domains, the route must specify the
+	// annotations specified in requiredAnnotations.  If domains is empty,
+	// the specified annotations are required for all newly created routes.
+	//
+	// +kubebuilder:validation:Optional
+	// +optional
+	Domains string `json:"domains,omitempty"`
```

Comment:
I can foresee cases where we might not want to apply attributes to a list of `Domain`s.  For example, we might want to exclude namespacea from domaina, and namespaceb from domainb.  To express that we need to have something like a map keyed on `Domain`, with `Policy` and `NamespaceSelector` values for each domain rather than a list of domains with a single policy or namespace selector that has to be applied to all.

Reply:
Or, the `IngressSpec.RequiredRouteAnnotations` expressed as a map with Domain as key?